CMakeLists.txt

```diff
@@ -25,6 +25,10 @@ SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror -Wall -Wextra -std=c99")
 
 SET(CMAKE_C_FLAGS_DEBUGWITHCOVERAGE "${CMAKE_C_FLAGS_DEBUGWITHCOVERAGE} -O0 -fprofile-arcs -ftest-coverage")
 
+find_program(MEMORYCHECK_COMMAND NAMES valgrind)
+set(MEMORYCHECK_COMMAND_OPTIONS
+        "--error-exitcode=5 --leak-check=full -v --track-origins=yes --partial-loads-ok=no")
```

Comment:
Could you add a short section to the readme indicating what valgrind is checking.  For folks who don't speak valgrind options, what sorts of errors, roughly, do we intend valgrind to catch?

Reply:
Good call.

Reply:
Good idea. Done.